File: FE/WEB/app/models/geo-region.js
import Model, { attr, hasMany } from '@ember-data/model';

export default class GeoRegionModel extends Model {
  @hasMany('geo-country', { async: true, inverse: 'geoRegion' })
  geoCountries;
  @hasMany('geo-subregion', { async: true, inverse: null })
  geoSubregions;

  @attr('string', { defaultValue: '' }) name;
  @attr('string', { defaultValue: '' }) translations;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/GeoRegion.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System.Collections.Generic;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    public class GeoRegion : Identifiable<int>
    {
        [Attr]
        public string Name { get; set; }
        [Attr]
        public string Translations { get; set; }

        [HasMany]
        public virtual ICollection<GeoSubregion> GeoSubregions { get; set; }
        [HasMany]
        public virtual ICollection<GeoCountry> GeoCountries { get; set; }
    }
}
